blocks/http/include/gnuradio-4.0/http/HttpBlock.hpp

```diff
@@ -0,0 +1,293 @@
+            while (!_shutdownThread) {
+                // long polling, just keep doing requests
+                std::thread thread{ &HttpBlock::doRequestEmscripten, this };
+                thread.join();
+            }
+        } else {
+            while (!_shutdownThread) {
+                while (_pendingRequests > 0) {
+                    _pendingRequests--;
+                    std::thread thread{ &HttpBlock::doRequestEmscripten, this };
+                    thread.join();
+                }
+                _ready.acquire();
+            }
+        }
+    }
+#else
+    void
+    runThreadNative() {
+        _client = std::make_unique<httplib::Client>(url);
+        _client->set_follow_location(true);
+        if (type == RequestType::SUBSCRIBE) {
+            // it's long polling, be generous with timeouts
+            _client->set_read_timeout(1h);
+            _client->Get(endpoint, [&](const char *data, size_t len) {
+                pmtv::map_t result;
+                result["mime-type"] = "text/plain";
+                result["status"]    = 200;
+                result["raw-data"]  = std::string(data, len);
+
+                queueWork(result);
+
+                return !_shutdownThread;
+            });
+        } else {
+            while (!_shutdownThread) {
+                while (_pendingRequests > 0) {
+                    _pendingRequests--;
+                    httplib::Result resp;
+                    if (type == RequestType::POST) {
+                        resp = parameters.empty() ? _client->Post(endpoint) : _client->Post(endpoint, parameters, "application/x-www-form-urlencoded");
+                    } else {
+                        resp = _client->Get(endpoint);
+                    }
+                    pmtv::map_t result;
+                    if (resp) {
+                        result["mime-type"] = "text/plain";
+                        result["status"]    = resp->status;
+                        result["raw-data"]  = resp->body;
+                        queueWork(result);
+                    }
+                }
+
+                _ready.acquire();
+            }
+        }
+    }
+#endif
+
+    void
+    queueWork(const pmtv::map_t &item) {
+        {
+            std::lock_guard lg{ _backlog_mutex };
+            _backlog.push(item);
+        }
+        const auto work = this->invokeWork();
+        if (work == work::Status::DONE) {
+            this->requestStop();
+        }
+        this->ioLastWorkStatus.exchange(work, std::memory_order_relaxed);
+    }
+
+    void
+    runThread() {
+#ifdef __EMSCRIPTEN__
+        runThreadEmscripten();
+#else
+        runThreadNative();
+#endif
+    }
+
+    void
+    startThread() {
+        lifecycle::State expectedThreadState = lifecycle::State::INITIALISED;
+        if (!this->_state.compare_exchange_strong(expectedThreadState, lifecycle::State::RUNNING, std::memory_order_acq_rel)) {
+            return;
+        }
+        this->_state.notify_all();
```

Comment:
I think this if statement is not needed and always return when using `Scheduler`. `_state` is always `RUNNING` at this point.

Reply:
Indeed, I removed it.